modin/core/io/io.py

```diff
@@ -277,6 +278,8 @@ def read_clipboard(cls, sep=r"\s+", **kwargs):  # pragma: no cover # noqa: PR01
     )
     def read_excel(cls, **kwargs):  # noqa: PR01
         ErrorMessage.default_to_pandas("`read_excel`")
+        if isinstance(kwargs["io"], ExcelFile):
+            kwargs["io"]._set_pandas_mode()
```

Comment:
this looks super-weird, could you please add an in-code comment explaining the _why this is needed_?